.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,48 @@
+name: Pull request
+
+on:
+  pull_request:
+    branches:
+      - develop
+
+jobs:
+  gradle:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Cheeckout
+      uses: actions/checkout@v2
+
+    - name: Set up JDK 17
+      uses: actions/setup-java@v2
+      with:
+        java-version: 17
+        distribution: 'temurin'
+        cache: 'gradle'
+
+    - name: Build with gradle
+      uses: gradle/gradle-build-action@v2
+      env:
+        HARVEST_ACCOUNT_ID_ADMIN1: ${{ secrets.HARVEST_ACCOUNT_ID_ADMIN1 }}
+        HARVEST_AUTH_TOKEN_ADMIN1: ${{ secrets.HARVEST_AUTH_TOKEN_ADMIN1 }}
+        HARVEST_ACCOUNT_ID_ADMIN2: ${{ secrets.HARVEST_ACCOUNT_ID_ADMIN2 }}
+        HARVEST_AUTH_TOKEN_ADMIN2: ${{ secrets.HARVEST_AUTH_TOKEN_ADMIN2 }}
+      with:
+        gradle-version: 7.2
+        arguments: build
+
+    - name: Generate test coverage report
+      run: |-
+        echo "Generating test coverage report"
+        ./gradlew jacocoTestReport
+
+    - name: Upload report to codecov.io
+      run: |-
+        echo "Uploading report..."
+        bash <(curl -s https://codecov.io/bash) -f build/reports/jacoco/test/jacocoTestReport.xml
+
+    - name: Save test results
```

Comment:
Does this work on GHA?

Reply:
Unfortunately not, I didn't manage to make it work with GHA, so I deleted this part.

Reply:
Are we talking about upload of artifacts, in this case the test results?
Yes, we this works on gha. Please take a look at [upload artifacts action](https://github.com/actions/upload-artifact).

Reply:
We wanted something like this:

https://github.com/marketplace/actions/publish-unit-test-results

But I couldn't make it work.
It was reporting 'No tests' each time.